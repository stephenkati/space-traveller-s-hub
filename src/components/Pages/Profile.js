import React from 'react';
import { useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';

const Profile = () => {
  const allMissions = useSelector((state) => state.mission);
  const joinedMissions = allMissions.missions.filter((mymission) => mymission.member);

  return (
    <section>
      <div className="missions-joined">
        <h1>My Missions</h1>
        {
          joinedMissions.length ? (
            <ul className="mission-joined-profile">
              {
                joinedMissions.map((mission) => (
                  <li key={mission.mission_id}>
                    {mission.mission_name}
                  </li>
                ))
              }
            </ul>
          ) : (
            <div>
              <p>No missions joined yet!</p>
              <span>
                Go to to
                <NavLink to="/missions"> Missions </NavLink>
                to join a mission
              </span>
            </div>
          )
        }
      </div>
    </section>
  );
};

export default Profile;
