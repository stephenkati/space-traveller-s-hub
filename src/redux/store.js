import { configureStore } from "@reduxjs/toolkit";
import MissionReducer from "./missions/MissionSlice";

const store = configureStore({
  reducer: {
    mission: MissionReducer,
  }
});

export default store;
