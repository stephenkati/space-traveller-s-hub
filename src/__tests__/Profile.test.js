import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Profile from '../components/Pages/Profile';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('tests for profile file', () => {
  test('snapshot test for profile component', () => {
    const component = render(
      <BrowserRouter>
        <Provider store={store}>
          <Profile />
        </Provider>
      </BrowserRouter>,
    );

    expect(component).toMatchSnapshot();
  });
});
