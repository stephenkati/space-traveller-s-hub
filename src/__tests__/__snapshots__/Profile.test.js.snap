// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for profile file snapshot test for profile component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="missions-joined"
        >
          <h1>
            My Missions
          </h1>
          <div>
            <p>
              No missions joined yet!
            </p>
            <span>
              Go to to
              <a
                class=""
                href="/missions"
              >
                 Missions 
              </a>
              to join a mission
            </span>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="missions-joined"
      >
        <h1>
          My Missions
        </h1>
        <div>
          <p>
            No missions joined yet!
          </p>
          <span>
            Go to to
            <a
              class=""
              href="/missions"
            >
               Missions 
            </a>
            to join a mission
          </span>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
